pkg/reconciliation/construct_statefulset_test.go

```diff
@@ -450,6 +451,44 @@ func Test_newStatefulSetForCassandraPodSecurityContext(t *testing.T) {
 	}
 }
 
+func TestValidName(t *testing.T) {
```

Comment:
These tests are good but if we have time to do it, it would be even better to have individual tests for `CleanupForKubernetes`, `CleanupSubdomain` and `CleanLabelValue` :-)

Reply:
I'll rename this to indicate it tests subdomain, but we do have those other tests. labels_test.go has one for CleanLabelValue and CleanupForKubernetes is tested in some other test as part of the process (but of course now I can't remember where - but at least Test_buildDefaultSuperuserSecret has a small test).